File: josephus.pl
#!/usr/bin/perl
use strict;
use warnings;
use Getopt::Long qw(GetOptions);

my (@arr);
my ($n,$k);

GetOptions(
    'n=i' => \$n,
    'k=i' => \$k,
) or die "Usage: $0 -n NUMBER -k NUMBER\n";

if ($n<=0 or $k<=0) { die "Arguments should be positive integers\n"; }

# Create array
for(my $i=1;$i<=$n;++$i) {
	push @arr,$i;
}
# Processing
for(my $count=0, my $j=1;@arr>1;++$count,++$j) {
	# At the end of array - go to beginning
	if ($count==$#arr+1) {
		$count=0;
	}
	# Remove POI, restart j, decrement count
	if ($j==$k) {
		splice(@arr,$count,1);
		$j=0;
		--$count;
	}
}
print "$arr[0]\n"; 

File: josephus.py
#!/usr/bin/python

import argparse
import sys

parser = argparse.ArgumentParser()
parser.add_argument('-n', action="store", dest='n', metavar="elements number", type=int)
parser.add_argument('-k', action="store", dest='k', metavar="step number", type=int)
args = parser.parse_args()

if args.n <= 0 or args.k <=0 :
    sys.exit('Arguments should be positive integers')

arr = []

# Fill array
i=0
while i<args.n :
    i += 1
    arr.append(i)
count=0
j=1
while len(arr) > 1 :
    # At the end of array - go to beginning
    if count==len(arr) :
        count=0
    # Remove POI, restart j, decrement count
    if j==args.k  :
        del arr[count]
        j=0
        count -= 1
    count += 1
    j += 1
print (arr[0])
